### PR TITLE
First step to enable Windows RyuJIT/arm32 testing

### PR DESCRIPTION
**Pull request by BruceForstall**
Still needs changes in code invoked by arm64_post_build.py to
set the COMPlus_AltJit variables correctly for RyuJIT.

### CODE REVIEW

**Comment by BruceForstall** [comment]
@jashook @RussKeldorph PTAL

I've renamed the existing 'arm' architecture to 'armlb' (for arm32 LEGACY_BACKEND testing) and made 'arm' now refer to RyuJIT/arm32.

This still needs some code somewhere to make the arm64 smarty invoker to set the complus variables correctly.

**Comment by jashook** [comment]
@dotnet-bot test CI Please

File: netci.groovy [hunk 1]
```diff
@@ -355,6 +355,9 @@ def static getJobName(def configuration, def architecture, def os, def scenario,
             }
             break
         case 'arm':
+            baseName = architecture.toLowerCase() + '_cross_' + configuration.toLowerCase() + '_' + os.toLowerCase()
+            break;
```
Comment (RussKeldorph) [inline netci.groovy]: nit: Semi-colon seems benign but inconsistent.

**Comment by BruceForstall** [comment]
@dotnet-bot test CI Please

**Comment by BruceForstall** [comment]
Hmmmm... well that's a new one. `Ubuntu16.04 arm Cross Debug Build` [link](https://ci.dot.net/job/dotnet_coreclr/job/master/job/arm_cross_debug_ubuntu16.04_prtest/5822/)
```
09:39:21 CMake Error at src/ToolBox/superpmi/superpmi-shim-simple/cmake_install.cmake:42 (file):
09:39:21   file INSTALL cannot copy file
09:39:21   "/opt/code/bin/obj/Linux.arm.Debug/src/ToolBox/superpmi/superpmi-shim-simple/libsuperpmi-shim-simple.so"
09:39:21   to "/opt/code/bin/Product/Linux.arm.Debug/./libsuperpmi-shim-simple.so".
09:39:21 Call Stack (most recent call first):
09:39:21   src/ToolBox/superpmi/cmake_install.cmake:41 (include)
09:39:21   src/ToolBox/cmake_install.cmake:38 (include)
09:39:21   src/cmake_install.cmake:52 (include)
09:39:21   cmake_install.cmake:44 (include)
09:39:21   
09:39:21 
09:39:21 
09:39:21 make: *** [install] Error 1
09:39:21 
Makefile:61: recipe for target 'install' failed
```

@dotnet-bot test Ubuntu16.04 arm Cross Debug Build

**Comment by BruceForstall** [comment]
@dotnet-bot test Ubuntu16.04 arm Cross Debug Build

**Comment by BruceForstall** [comment]
@dotnet-bot test CI Please

**Comment by BruceForstall** [comment]
@dotnet-bot test CI Please

**Comment by BruceForstall** [comment]
@dotnet-bot test CentOS7.1 x64 Debug Build and Test
@dotnet-bot test Tizen armel Cross Debug Build
@dotnet-bot test Ubuntu x64 Checked Build and Test
@dotnet-bot test Windows_NT arm Cross Checked Build and Test

**Comment by BruceForstall** [comment]
@dotnet-bot test CI Please

**Comment by BruceForstall** [comment]
@dotnet-bot test CI Please

**Comment by BruceForstall** [comment]
@dotnet-bot test CI Please

**Comment by BruceForstall** [comment]
@jashook The generator_prtest jobs have passed.

**Comment by jashook** [comment]
lgtm

**Comment by jashook** [comment]
Feel free to merge,

**Comment by jashook** [comment]
I have started the following jobs. Will be monitoring them to make sure everything is as expected.

https://ci.dot.net/job/dotnet_coreclr/job/master/job/arm64_cross_checked_windows_nt/2498/
https://ci.dot.net/job/dotnet_coreclr/job/master/job/arm_cross_checked_windows_nt_prtest/651
https://ci.dot.net/job/dotnet_coreclr/job/master/job/armlb_cross_checked_windows_nt_prtest/1

Note some of these jobs are still being scheduled. Links will be available after that happens.

**Pull request merged**